.github/workflows/integration_test_run.yaml

```diff
@@ -261,8 +258,15 @@ jobs:
         if: ${{ github.event_name == 'pull_request' }}
         continue-on-error: true
         with:
-          name: ${{ env.CHARM_NAME }}-charm
-          path: ${{ inputs.working-directory }}
+          name: charm-artifacts
+      - name: Add charm-file arguments
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          args=${{ env.ARGS }}
+          for charm in $(ls *.charm); do
+            args="${args} --charm-file=${charm}"
```

Comment:
I think `*.charm` will only list the charm files in the current directory, and `**/*.charm` is to recursively list all charm files? 
Another question is that if we are using the same argument construction `args="${args} --charm-file=${charm}"`, which would produce something like `--charm-file=aaa/aaa.charm --charm-file=bbb/bbb.charm --charm-file=ccc/ccc.charm`, then how does the integration know which charm to use?

Reply:
Gotcha, if that's the case, I think we can remove the `build-charm` job and related steps, since that's just `build-charms` but with only one charm.

Reply:
yeah, i was trying to think of a way to not break other users of these workflows.  That's why I only ran `build-charms` job if `build-charm` didn't produce a `charm-file`

if there are others depending on these artifacts -- then it could be an issue.  

It's also worth noting i didn't take any care to build rocks or images for the  multiple charms case -- nor did i try to find a way to distinctly manage which charm needs which image. 

I'm not totally sure this PR contains all the right steps needed to manage these complexities. Perhaps it should be closed and I try again with a spec. 

It won't be long to where I need multiarch charms.  This will produce multiple `.charm` files for each supported architecture.  Each of these will need to be published associated with the respective image resources for those architectures.  Again -- another reason for perhaps stepping back and making a spec.

Reply:
Oh, yes, that reminds me that the publish workflow actually uses the old charm artifact. I think the entire `operator-workflows` will be rewritten in the future, but I think I am good with the current status. Thanks!